shared/Microsoft.Extensions.Certificates.Generation.Sources/CertificateManager.cs

```diff
@@ -0,0 +1,414 @@
+
+        public const string AspNetIdentityOid = "1.3.6.1.4.1.311.84.1.2";
+        public const string AspNetIdentityOidFriendlyName = "Asp.Net Core Identity Json Web Token signing development certificate";
+
+        private const string ServerAuthenticationEnhancedKeyUsageOid = "1.3.6.1.5.5.7.3.1";
+        private const string ServerAuthenticationEnhancedKeyUsageOidFriendlyName = "Server Authentication";
+
+        private const string LocalhostHttpsDnsName = "localhost";
+        private const string LocalhostHttpsDistinguishedName = "CN=" + LocalhostHttpsDnsName;
+
+        private const string IdentityDistinguishedName = "CN=Microsoft.AspNetCore.Identity.Signing";
+
+        public const int RSAMinimumKeySizeInBits = 2048;
+
+        public IList<X509Certificate2> ListCertificates(
+            CertificatePurpose purpose,
+            StoreName storeName,
+            StoreLocation location,
+            bool isValid)
+        {
+            var certificates = new List<X509Certificate2>();
+            try
+            {
+                using (var store = new X509Store(storeName, location))
+                {
+                    store.Open(OpenFlags.ReadOnly);
+                    certificates.AddRange(store.Certificates.OfType<X509Certificate2>());
+                    IEnumerable<X509Certificate2> matchingCertificates = certificates;
+                    switch (purpose)
+                    {
+                        case CertificatePurpose.All:
+                            matchingCertificates = matchingCertificates
+                                .Where(c => HasOid(c, AspNetHttpsOid) || HasOid(c, AspNetIdentityOid))
+                                .ToList();
+                            break;
+                        case CertificatePurpose.HTTPS:
+                            matchingCertificates = matchingCertificates
+                                .Where(c => HasOid(c, AspNetHttpsOid))
+                                .ToList();
+                            break;
+                        case CertificatePurpose.Signing:
+                            matchingCertificates = matchingCertificates
+                                .Where(c => HasOid(c, AspNetIdentityOid))
+                                .ToList();
+                            break;
+                        default:
+                            break;
+                    }
+                    if (isValid)
+                    {
+                        // Ensure the certificate hasn't expired, has a private key and its exportable
+                        // (for container/unix scenarios).
+                        var now = DateTimeOffset.Now;
+                        matchingCertificates = matchingCertificates
+                            .Where(c => c.NotBefore <= now &&
+                                now <= c.NotAfter &&
+                                ((c.GetRSAPrivateKey() is RSACryptoServiceProvider rsaPrivateKey &&
```

Comment:
Can you add a flag to make the exportability check optional (and instead just check that the cert has a private key)?